src/tests/stdlib.rs

```diff
@@ -418,3 +418,107 @@ fn lines() {
     ";
     test_amber!(code, "line: hello\nline: world")
 }
+
+#[test]
+fn is_command() {
+    let code = "
+        import { is_command } from \"std\"
+        main {
+            if is_command(\"cat\") {
+                echo \"exist\"
+            }
+        }
+    ";
+    test_amber!(code, "exist")
+}
+
+#[test]
+fn create_symbolic_link() {
+    let code = "
+        import { create_symbolic_link } from \"std\"
+        main {
+            unsafe $touch /tmp/amber-symbolic$
+            if create_symbolic_link(\"/tmp/amber-symbolic\", \"/tmp/amber-symbolic-link\") {
+                echo \"created\"
+            } else {
+                echo \"failed\"
+            }
+            unsafe $rm /tmp/amber-symbolic$
+            unsafe $rm /tmp/amber-symbolic-link$
+        }
+    ";
+    test_amber!(code, "created")
+}
+
+#[test]
+fn create_dir() {
+    let code = "
+        import { create_dir, dir_exist } from \"std\"
+        main {
+            create_dir(\"/tmp/amber-test\")
+            if dir_exist(\"/tmp/amber-test\") {
+                unsafe $rm /tmp/amber-test$
+                echo \"created\"
+            }
+        }
+    ";
+    test_amber!(code, "created")
+}
+
+#[test]
+fn make_executable() {
+    let code = "
+        import { make_executable } from \"std\"
+        main {
+            unsafe $touch /tmp/amber-symbolic$
+            if make_executable(\"/tmp/amber-symbolic\") {
+                echo \"created\"
+            }
+            unsafe $rm /tmp/amber-symbolic$
+        }
+    ";
+    test_amber!(code, "created")
+}
+
+#[test]
+fn switch_user_permission() {
+    // We use `whoami` to get the running user to assign again the same user as permission
+    let code = "
+        import { switch_user_permission } from \"std\"
+        main {
+            unsafe $touch /tmp/amber-symbolic$
+            if switch_user_permission(unsafe $whoami$,\"/tmp/amber-symbolic\") {
+                echo \"done\"
+            }
+            unsafe $rm /tmp/amber-symbolic$
+        }
+    ";
+    test_amber!(code, "done")
+}
+
+#[test]
+fn download() {
+    let code = "
+        import { download, is_command } from \"std\"
+        main {
+            if download(\"https://amber-lang.com/\", \"/tmp/amber-homepage\") {
```

Comment:
what if there is no network connection? or if the site is down?

Reply:
> So I think that we have no choice for this test...

it could spin up a simple http server in a separate thread and send a request to there

Reply:
wow I saw the commit, cool solution :-)